Dev tools with the usual suspects

renatodeleao opened the pull request:
Fixes #1 , #3 and almost #4* 
Mostly from: https://github.com/renatodeleao/gulp-babel-boilerplate

- Gulp: sass, jade, starting web, deploys, minification.
- Webpack: bundle our lib and site scripts.
- Right now bundles site to `/docs` to be available on github pages,
- bundle library to `dist/` 

* I can't  export it so it is available as
```javaScript 
import Flooper from './Flooper.js'
```
@jlbribeiro ?

So right now I expose it as a global variable named `Flooper`. Yeah shame on me. Webpack allegedly does the magic by adding this, but it's no working.
```javascript
output:{
    // export itself to a global var
    libraryTarget: "var",
    // name of the global var: "Foo"
    library: "Flooper"
}
```



Todos for another PR:
- Remove hardcodedpaths
- Move site from `/docs` to an orphan `gh-pages` branch (use task from `qnorr`).

renatodeleao committed:
Gulp & Webpack

because why not?
From: https://github.com/renatodeleao/gulp-babel-boilerplate

- Gulp: sass, jade, starting web, deploys, magnification.
- Webpack: bundle our lib and site javascripts

Right now bundles site to docs to be available on github pages

bundle library to dist/ folder (and to docs so it can be used when
deployed)

PS: Yes they can live well together

renatodeleao committed:
Build

In the future only library builds will be committed to the repo

renatodeleao committed:
updated readme

tomasmcm commented on site/stylesheets/index.scss:
```diff
@@ -80,6 +81,8 @@ h1 {
 
 .c-infinity__block {
+	margin: 0 15px;
+	padding: 0 15px;
```
These two lines should not be in this pull request... But since there is already a new one with the fix for the jumps, its ok

renatodeleao replied:
True. 😴

tomasmcm commented on docs/index.html:
```diff
@@ -1,41 +1,39 @@
+    <meta name="description" content="Desc">
+    <meta name="keywords" content="gulp, babel, sass, jade, javaScript">
+    <title>Flooper.js</title>
+    <meta name="viewport" content="width=device-width,initial-scale=1.0,minimum-scale=1.0,maximum-scale=1.0,user-scalable=no">
```
Wouldn't it be nice, if one day... you know... we didn't had to add 100000000 lines in the head tag every time we create a new doc? 🤔 💭

renatodeleao replied:
We can, we are using **Jade** templating language. (don't ask me why, saw it, on other starter kit and it simple enough, maybe not as powerful as `panini` which we use on `qnorr` but enough for a single page.

`docs/*` are generated files. Everything comes from `./site` folder!

Bonus tip: all meta info comes from `jade/config.json`, that are exposed to `.jade` files as variables 🙌 
Bonus tip2: it supports partials, just go to `site/jade/` and create `_head.jade`. 🎩

tomasmcm reviewed (approved):
Nice work, :shipit:

tomasmcm commented on gulpfile.babel.js:
```diff
@@ -0,0 +1,232 @@
+      $.watch("./site/sass/**/*", () => {
+        gulp.start("sass");
+      });
+
```
What about listening for changes on the source file?
```
$.watch("./lib/**/*", () => {
    gulp.start("build");
});
```

renatodeleao replied:
@tomasmcm everything related to javascript is done by webpack. `Check webpack.config.babel.js`. What we can do and you're right is to reload browser to reflect lib changes (not sure if not already doing it)

```javascript
$.watch("./lib/**/*", () => {
    bs.reload() // reload browserSync; // while we don't hot module reloading.
});
```

tomasmcm replied:
@renatodeleao that does not work. How about this:
```
$.watch("./lib/**/*", () => {
    gulp.start("webpack");
});
```

renatodeleao replied:
Sry mate, I was being stupid.

Yes just is like you said above, but passing stream paths as an array, instead of duplicating the watch function like this:

```javascript
$.watch(["./site/js/**/*", "./lib/**/*" ], () => {
  gulp.start("webpack");
});
```

Will commit an update. Thanks :thumbs_up:

tomasmcm committed:
fix path name

renatodeleao committed:
Watch for changes on lib files and reload

renatodeleao committed:
Updated urls to Whitesmith

renatodeleao committed:
:skull: to hardcoded paths && docs folder

- All paths defined in package.json, can be imported to both gulp and
webpack and also handy to use variables in pm scripts!

- we will use gh-pages orphan branch for the site.

- site is served on public/ folder and this folder is also used to
deploy it.

- add npm scripts to build and deploy

- goodby docs branch

renatodeleao commented:
@tomasmcm I'll shipit. @jlbribeiro cagou-me na boca so i'll open a new issue to tackle the problem
